pkg/apiserver/registry/intelligence/networkpolicyrecommendation/rest.go

```diff
@@ -153,11 +180,34 @@ func (r *REST) copyNetworkPolicyRecommendation(intelli *intelligence.NetworkPoli
 	intelli.Status.SparkApplication = crd.Status.SparkApplication
 	intelli.Status.CompletedStages = crd.Status.CompletedStages
 	intelli.Status.TotalStages = crd.Status.TotalStages
-	if crd.Status.RecommendedNP != nil {
-		intelli.Status.RecommendedNetworkPolicy = crd.Status.RecommendedNP.Spec.Yamls
-	}
 	intelli.Status.ErrorMsg = crd.Status.ErrorMsg
 	intelli.Status.StartTime = crd.Status.StartTime
 	intelli.Status.EndTime = crd.Status.EndTime
 	return nil
 }
+
+func (r *REST) getRecommendationResult(id string) (result string, err error) {
```

Comment:
This is more a design question, but do you think there is a need to query clickhouse for every GET? Or once the results are populated in the resource status we can skip the CH query?

The results are kind of static, unless I'm missing something obvious we will get the same data for every query...

Reply:
Yes, I think this is a tradeoff between space and time. As we adopt the structured data in ClickHouse and remove RecommendedNetworkPolicy CRD, I think we do not plan to save the data both in ClickHouse and etcd? I also hesitate with saving the same data from database in the API server. Do you prefer to have a simple cache in the API server to avoid the repeated GET?

Reply:
I believe you are correct - and your explanation makes perfect sense